x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+restart the node.
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
+keystores that are currently in use. You'll use the same names for your new
+certificate.
++
+In this example, the keystore and truststore are pointing to different files.
+Your configuration might use the same file that contains the certificate and CA.
+In this case, include the path to that file for both the keystore and truststore.
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/elastic-certificates.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. On every node in your cluster, generate a certificate and private key for your
+node. You must include the CA that was used to sign the certificate currently
+in use.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `--ca <ca_file>`:: Name of the CA keystore used to sign your certificates. The
+   default file name from the `elasticsearch-certutil` tool is
+   `elastic-stack-ca.p12`.
+====
++
+    a. Enter the password for your CA keystore when prompted.
+
+    b. Enter the name for your keystore, which must match the name of the
+    keystore currently in use.
++
+NOTE: If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename. 
+
+    c. (Optional) Enter a password for your keystore. 
+
+. If you entered a password when creating the node certificate, run the
+following command to store the password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+--
+
+. Stop {es} on the first node in your cluster where you are updating the
+keystore.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `elastic-stack-ca.p12`.
+
+. If your keystore is password protected, the password is stored in the {es}
+keystore, _and_ the password needs to change, then update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore
+file that you created.
+
+. Restart the node where you updated the keystore. 
+
+. (Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. On each additional node in your cluster, stop {es}, update the keystore,
+and restart {es}.
+
+[[node-certs-same-http]]
+==== Generate a new certificate for the HTTP layer
+The certificate for the HTTP layer is used to verify other components when they
+connect to {es}, such as {kib} or any of the Elastic language clients.
+
+. Stop {es} and {kib} on the first node in your cluster where you are updating
```

Comment:
kibana and es won't necessarily be on the same node. Let's break this into 2 steps, one for kibana and one for es

Reply:
Also, we do we suggest that they stop kibana ?

Reply:
I thought that we had to stop Kibana. Now that I think more, that might only be necessary if users are updating the certificate for [encrypting HTTP communications for Kibana](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-basic-setup-https.html#encrypt-kibana-http). Is that worth calling out here?

Reply:
If ES has TLS enabled for HTTP ( which is an assumption here because we are rotating HTTP TLS certificates ) then Kibana has to trust this certificate when communicating with ES. The common way to do this in Kibana ( and other clients, and TLS in general ) is to trust the CA certificate that has signed a certificate( in kibana the setting is `elasticsearch.ssl.certificateAuthorities:` ) . 

In the case that we are reusing the CA , Kibana needs no update to its config, thus no need to stop it. 

In the case that we are using a new CA, Kibana configuration needs to be updated to trust the new CA. We should check with kibana-security what is the best way for the configuration to be updated in kibana and whether or not it requires kibana to be restarted and incorporate this here

Reply:
@legrego, can we get your input here? This PR adds documentation for [updating TLS certificates](https://elasticsearch_73781.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/update-node-certs.html), and I'm wondering if we also need to update the cert for encrypting HTTP communications for Kibana. If so, what's the best way to update the configuration and where in this process should users stop/restart Kibana?

Reply:
If the CA is changing, then Kibana should be updated to trust the new CA, which requires a restart. The way this happens depends on the format of the CA: we accept either PEM or PKCS#12, and the way to configure these is outlined in **step 3** here: https://www.elastic.co/guide/en/kibana/current/configuring-tls.html#configuring-tls-kib-es

The recommended order of operations for Kibana is:
1) Update the configuration
2) Restart Kibana

You could stop Kibana beforehand if you want to -- either approach would be fine.

FWIW, we have an open issue to support CA/cert updates without a restart, but it's not something we offer today: https://github.com/elastic/kibana/issues/101072

Reply:
Thanks @legrego -- I added instructions for updating the Kibana certificate in c855d0e8fb195f51037d685fab4535f200c23363